backend/nfconfig/config.go

```diff
@@ -124,11 +130,73 @@ func sortPlmnSnssaiConfig(plmnSnssai []nfConfigApi.PlmnSnssai) {
 	}
 }
 
-func (c *inMemoryConfig) syncAccessAndMobility() {
-	c.accessAndMobility = []nfConfigApi.AccessAndMobility{}
+func (c *inMemoryConfig) syncAccessAndMobility(networkSlices []configmodels.Slice) {
+	plmnSnssaiTacsMap := map[accessAndMobilityKey][]string{}
+	for _, s := range networkSlices {
+		accessAndMobilityTmp := accessAndMobilityKey{
+			plmn:    s.SiteInfo.Plmn,
+			sliceId: s.SliceId,
+		}
+		if plmnSnssaiTacsMap[accessAndMobilityTmp] != nil {
+			logger.NfConfigLog.Warnf("Found duplicate Network slice `%+v` for PLMN `%+v`, merging TACs for Access and Mobility", s.SliceId, s.SiteInfo.Plmn)
+		} else {
+			plmnSnssaiTacsMap[accessAndMobilityTmp] = []string{}
+		}
+		for _, g := range s.SiteInfo.GNodeBs {
+			tac := strconv.Itoa(int(g.Tac))
+			if !slices.Contains(plmnSnssaiTacsMap[accessAndMobilityTmp], tac) {
+				plmnSnssaiTacsMap[accessAndMobilityTmp] = append(plmnSnssaiTacsMap[accessAndMobilityTmp], tac)
+			}
+		}
+	}
+	c.accessAndMobility = convertPlmnSnssaiTacsMapToSortedList(plmnSnssaiTacsMap)
 	logger.NfConfigLog.Debugln("Updated Access and Mobility in-memory configuration. New configuration:", c.accessAndMobility)
 }
 
+func convertPlmnSnssaiTacsMapToSortedList(plmnSnssaiMap map[accessAndMobilityKey][]string) []nfConfigApi.AccessAndMobility {
+	newAccessAndMobilityConfig := []nfConfigApi.AccessAndMobility{}
+	for plmnSliceId, tacList := range plmnSnssaiMap {
+		plmnId := nfConfigApi.NewPlmnId(plmnSliceId.plmn.Mcc, plmnSliceId.plmn.Mnc)
+		parsedSnssai, err := parseSnssaiFromSlice(plmnSliceId.sliceId)
+		if err != nil {
+			logger.NfConfigLog.Warnf("Error in parsing SNSSAI: %v. Network slice `%+v` will be ignored", err, plmnSliceId.sliceId)
+			continue
+		}
+		accessAndMobility := nfConfigApi.NewAccessAndMobility(*plmnId, parsedSnssai)
+		accessAndMobility.Tacs = tacList
+		newAccessAndMobilityConfig = append(newAccessAndMobilityConfig, *accessAndMobility)
+	}
+	sortAccessAndMobilityConfig(newAccessAndMobilityConfig)
+	return newAccessAndMobilityConfig
+}
+
+func sortAccessAndMobilityConfig(accessAndMobility []nfConfigApi.AccessAndMobility) {
+	sort.Slice(accessAndMobility, func(i, j int) bool {
+		if accessAndMobility[i].PlmnId.GetMcc() != accessAndMobility[j].PlmnId.GetMcc() {
+			return accessAndMobility[i].PlmnId.GetMcc() < accessAndMobility[j].PlmnId.GetMcc()
+		}
+		if accessAndMobility[i].PlmnId.GetMnc() != accessAndMobility[j].PlmnId.GetMnc() {
+			return accessAndMobility[i].PlmnId.GetMnc() < accessAndMobility[j].PlmnId.GetMnc()
+		}
+		if accessAndMobility[i].Snssai.GetSst() != accessAndMobility[j].Snssai.GetSst() {
+			return accessAndMobility[i].Snssai.GetSst() < accessAndMobility[j].Snssai.GetSst()
+		}
+		if !accessAndMobility[i].Snssai.HasSd() && accessAndMobility[j].Snssai.HasSd() {
+			return true
+		}
+		if accessAndMobility[i].Snssai.HasSd() && !accessAndMobility[j].Snssai.HasSd() {
+			return false
```

Comment:
Why not just doing the following?
```suggestion
               if accessAndMobility[i].Snssai.HasSd() != accessAndMobility[j].Snssai.HasSd() {
                        return !accessAndMobility[i].Snssai.HasSd()
```